### PR TITLE
operator: updated logic around managing secret config to also consider additional scrape targets

### PR DESCRIPTION
**Pull request by mjpitz**
Resolves gh-2840

### CODE REVIEW

File: pkg/prometheus/operator.go [hunk 1]
```diff
@@ -1076,9 +1076,9 @@ func (c *Operator) sync(key string) error {
 		return err
 	}
 
-	// If no service monitor selectors are configured, the user wants to
-	// manage configuration themselves.
-	if p.Spec.ServiceMonitorSelector != nil {
```
Comment (vsliouniaev) [inline pkg/prometheus/operator.go]: Would this mean that podSelectors suffer from the same issue if they are the only configuration set?

Comment (mjpitz) [reply pkg/prometheus/operator.go]: Yeah. After getting some feedback on the original ticket, we decided to remove this gate entirely.

**Comment by brancz** [comment]
PromReloadConfig is the e2e test that failed. It might work with the old “feature”. We should double check that and make it use additional scrape configs if so.

**Comment by mjpitz** [comment]
So trying to run these tests locally, but I'm hitting an issue where it can't pull the config reloader "latest" tag. I took a look at quay and it doesn't seem like there is a "latest" tag there.

**Comment by mjpitz** [comment]
Does anyone have any feedback on how I can go about testing this?

**Comment by brancz** [comment]
Does the [testing section](https://github.com/coreos/prometheus-operator#testing) in the readme help?

**Comment by mjpitz** [comment]
@brancz : That's how I wound up at the image pull issue seen [above](https://github.com/coreos/prometheus-operator/pull/2842#issuecomment-553696377). I event tried pulling the image directly. Digging through the code, I did see this image referenced in some statefulset tests, but when I try to pull the image, it fails.

```
docker pull quay.io/coreos/prometheus-config-reloader:latest
Error response from daemon: manifest for quay.io/coreos/prometheus-config-reloader:latest not found: manifest unknown: manifest unknown
```

**Comment by brancz** [comment]
Try running `make clean image` before running the e2e tests.

**Comment by brancz** [comment]
When running that you should get a build output similar to this:

```shell
$ make image
GOOS=linux CGO_ENABLED=0 go build -mod=vendor -ldflags="-s -X github.com/coreos/prometheus-operator/pkg/version.Version=0.34.0" -o operator cmd/operator/main.go
docker build -t quay.io/coreos/prometheus-operator:354da880 .
Sending build context to Docker daemon  182.7MB
Step 1/4 : FROM quay.io/prometheus/busybox:latest
 ---> a130381627cc
Step 2/4 : ADD operator /bin/operator
 ---> 925ff0357ee5
Step 3/4 : USER 65534
 ---> Running in 5dd4f921fa92
Removing intermediate container 5dd4f921fa92
 ---> 206a7fe8ed05
Step 4/4 : ENTRYPOINT ["/bin/operator"]
 ---> Running in 27f44c4cb588
Removing intermediate container 27f44c4cb588
 ---> fb865ece34ab
Successfully built fb865ece34ab
Successfully tagged quay.io/coreos/prometheus-operator:354da880
touch .hack-operator-image
GOOS=linux CGO_ENABLED=0 go build -mod=vendor -ldflags="-s -X github.com/coreos/prometheus-operator/pkg/version.Version=0.34.0" -o prometheus-config-reloader cmd/prometheus-config-reloader/main.go
docker build -t quay.io/coreos/prometheus-config-reloader:354da880 -f cmd/prometheus-config-reloader/Dockerfile .
Sending build context to Docker daemon  190.2MB
Step 1/4 : FROM quay.io/prometheus/busybox:latest
 ---> a130381627cc
Step 2/4 : ADD prometheus-config-reloader /bin/prometheus-config-reloader
 ---> 34e2ceb0365e
Step 3/4 : USER nobody
 ---> Running in d4e561bdd160
Removing intermediate container d4e561bdd160
 ---> 8da4f5cce24e
Step 4/4 : ENTRYPOINT ["/bin/prometheus-config-reloader"]
 ---> Running in 98bc3eb941ec
Removing intermediate container 98bc3eb941ec
 ---> ea3735c4db47
Successfully built ea3735c4db47
Successfully tagged quay.io/coreos/prometheus-config-reloader:354da880
touch .hack-prometheus-config-reloader-image
```

It builds images with your local docker host with the tag being the short git tag. Subsequently running `make test-e2e` uses the same tag. I'm not sure where the `latest` tag is coming from for you. When I run it it runs the git tag as expected.

**Comment by mjpitz** [comment]
Closing this for now.. Probably easier to re-patch.

**Pull request closed without merging**